### ISSUE TITLE
In lieu of complete automation, provide better documentation

### ISSUE THREAD

**Issue by geerlingguy** [geerlingguy/ansible-role-certbot#6]
I'm leery of letting automated software that's out of my control from managing my own config files, especially when it comes to crypto. While many people are better off letting LE/Certbot do everything for them (many of these people hand-edit config files on their servers—gasp!), I'd rather let Certbot work on cert redemption/renewal, and then I'll work on the actual Apache config.

I'd like to document my process, at least, so I can see where it's more automatable... and so others can see how simple/better the process is for further automation if we use Ansible to the greatest extent possible.

**Comment by geerlingguy** [comment]
Right now my process is:

1. Provision new server using this role (`geerlingguy.certbot`), and set up Apache/Nginx config to direct traffic to port 80 for each virtualhost.
2. Log in, run (not as root): `/path/to/certbot-auto --apache certonly -d [domain] -d [domain]` (for all the domains on the server)
3. Run through the certbot UI answering questions, setting up the account, etc.
4. Once the certs are generated, edit my Apache/Nginx config to redirect 80 to 443 with the following configuration per virtualhost (example is on Ubuntu):

**/etc/apache2/ssl.conf**:

```
SSLEngine on
SSLCipherSuite ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:AES:CAMELLIA:DES-CBC3-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!EDH-RSA-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA
SSLProtocol -All +TLSv1 +TLSv1.1 +TLSv1.2
SSLHonorCipherOrder On
SSLCompression off
```

**/etc/apache2/sites-enabled/[site].conf**:

```
<VirtualHost *:80>
  ServerName www.example.com
  Redirect permanent / https://www.example.com/
</VirtualHost>
<VirtualHost *:443>
  ServerName www.example.com
  ServerAlias example.com
  ServerAdmin webmaster@example.com
  DocumentRoot /path/to/example_com

  Include /etc/apache2/ssl.conf
  SSLCertificateFile /etc/letsencrypt/live/www.example.com/fullchain.pem
  SSLCertificateKeyFile /etc/letsencrypt/live/www.example.com/privkey.pem
  SSLCertificateChainFile /etc/letsencrypt/live/www.example.com/chain.pem

  <Directory "/path/to/example_com">
    AllowOverride All
    Options -Indexes +SymLinksIfOwnerMatch
    Order deny,allow
    Require all granted
  </Directory>
</VirtualHost>
```

This example is Ubuntu / Apache; Nginx is similar but with a different configuration layout.

**Comment by llbbl** [comment]
So the "privkey.pem" is updated every time renew is run with the current one?  The directory with all the keys (/etc/letsencrypt/keys) just keeps adding more. This was a non-obvious realization that I thought was super important to not breaking the site every time 'renew' is ran.

**Comment by geerlingguy** [comment]
For Nginx:

  - In step 2, use `--nginx` instead of `--apache`.
  - TODO.

**Comment by geerlingguy** [comment]
Added some more docs in the README based on work I'm doing for a separate project.

**Comment by geerlingguy** [comment]
See also, #12 — I'm doing a little testing with this for at least Nginx (and probably will work on Apache as well).

**Comment by geerlingguy** [comment]
tl;dr: https://github.com/geerlingguy/ansible-role-certbot/issues/12#issuecomment-350941154

Closing this ticket in favor of follow-up to automate generation using --webroot (PR https://github.com/geerlingguy/ansible-role-certbot/pull/38 already adds --standalone automated cert generation support.)
